modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/artifact/MavenClassLoaderConfigurationLoader.java

```diff
@@ -68,10 +75,11 @@ private void refresh() {
   }
 
   private void createClassLoaderConfigurationLoaders() {
-    Optional<MavenClient> mavenClient = ofNullable(createMavenClient(mavenRuntimeConfig));
+    closeMavenClient();
```

Comment:
is this something that may happen?

Reply:
It could happen if the maven configuration changed between invocations to the `load` method.